crates/net/src/builder.rs

```diff
@@ -193,6 +202,14 @@ impl NetworkDriverBuilder {
     }
 }
 
```

Comment:
I think the `socket` and `listen_config` are the same config, user should only set one of them, right? If `socket` not set, we must construct a `NetworkAddress` from `listen_config`, right?
If above are right, it seems we must create a `NetworkAddress` object for creating `GossipDriver` so that `DiscoveryBuilder` should always have it. The `listen_config` for `DiscoveryBuilder` seems not necessary if no other caller.

Reply:
The core idea here was that the [`ListenConfig`](https://docs.rs/discv5/latest/discv5/socket/enum.ListenConfig.html) can be used to create a dual stack whereas we cannot do that with the `SocketAddr`.

I've just opened a PR to actually remove the `NetworkAddress` since it just acts as a conversion between `SocketAddr` and `MultiAddr`: https://github.com/paradigmxyz/op-rs/pull/55.

I think what you have here is good - we expose both `ListenConfig` and `SocketAddr` builder setters.

- The `SocketAddr` is always used for the `DiscoveryBuilder`.
- The `ListenConfig` is used for the `GossipDriver` and if it isn't set, we convert the socket addr into a listen config.
- If the socket address isn't set, we error.